pkg/cmd/pipeline/start.go

```diff
@@ -152,6 +153,7 @@ like cat,foo,bar
 	c.Flags().StringVarP(&opt.PrefixName, "prefix-name", "", "", "specify a prefix for the pipelinerun name (must be lowercase alphanumeric characters)")
 	c.Flags().StringVarP(&opt.TimeOut, "timeout", "", "1h", "timeout for pipelinerun")
 	c.Flags().StringVarP(&opt.Filename, "filename", "f", "", "local or remote file name containing a pipeline definition to start a pipelinerun")
+	c.Flags().BoolVarP(&opt.ParamUseDefaults, "params-use-defaults", "", false, "allow using deafult parameter values without prompting for input")
```

Comment:
I wonder if a flag-based approach is the right way to go about this. Maybe what the better approach would be is to not prompt the user at all if a default is specified for a param and the user specifies nothing, and prompt if there is no default value and nothing is specified via -p. 

cc @piyush-garg

Reply:
@danielhelfand , Yes, I agree with you, I too thought so, but I thought that would be something which will be removed(removal of prompting), and I think the members should decide that. If you want I can change the PR to do that.

Reply:
> I wonder if a flag-based approach is the right way to go about this. Maybe what the better approach would be is to not prompt the user at all if a default is specified for a param and the user specifies nothing, and prompt if there is no default value and nothing is specified via `-p`.
> 
> cc @piyush-garg

Even if we would like that approach, this cannot change in one go (we need to warn of the default behavior change, wait for a release or too and then switch). 
To be honest, I do like the fact that, by default, it asks me for values even if they have a default. It feels more user friendly for me, as it lets me override them if I need.

/cc @chmouel @pradeepitm12

Reply:
I think that was a @siamaksade request to have default...

Reply:
Yes, I do appreciate interactive mode here and allowing the user flexibility as far as picking a default or overriding it as well. I think the main thing would be with whatever approach is recommended is that a user doesn't have to only use defaults in a non-interactive scenario. 

Maybe a list of params with defaults could be passed to pick which param defaults to use via a flag? We could also support an all param defaults flag as well? What I want to make sure of is that a user can pick certain param defaults to use while also overriding params that have defaults.

Reply:
@danielhelfand , One approach n mind is -
- If a param having default configured and not been specified with -p should take the default value.
- If a param having default configured and its value is specified with -p should override the default value.
- if any param does not have a default configured and its value is not specified with -p should give an interactive prompt to give the value.

What do you guys say? @vdemeester @chmouel

Reply:
I find it confusing not to prompt user for parameters that have a default value, unless user specifically asks for this e.g. via a flag. "-p" should be the only indicator for not promoting otherwise IMO user should still get prompted but the default value should be displayed (current behaviour) so they can press enter and continue.

Reply:
I think the idea of the flag in this pr would be to not have to use `-p` to use param defaults while also allowing params with defaults to still be overrode using `-p`. Basically the idea would be to have a shortcut to specifying to use default values as opposed to having to use `-p` when the default is to be used. I think the idea makes sense from that perspective. And then prompting would remain exactly as is.

Reply:
@danielhelfand , yes, that is the idea .

Reply:
Yup, sorry for the confusion in the beginning. Glad we talked it out though and have clarity that this is the right approach.